app/variablesmanager.cpp

```diff
@@ -0,0 +1,41 @@
+#include "variablesmanager.h"
+
+#include "qgsexpressioncontextutils.h"
+
+VariablesManager::VariablesManager( MerginApi *merginApi, QObject *parent )
+  : QObject( parent )
+  , mMerginApi( merginApi )
+{
+  QgsExpressionContextUtils::setGlobalVariable( QStringLiteral( "mergin_url" ),  mMerginApi->apiRoot() );
```

Comment:
Is it always guaranteed that authChanged() is called? e.g. if user opens Input with a project loaded without going to Projects page, is the username going to be set?